tests/suite/fixtures.py

```diff
@@ -727,6 +733,64 @@ def fin():
     )
 
 
+class TransportServerSetup:
+    """
+    Encapsulate Transport Server Example details.
+
+    Attributes:
+        name (str):
+        namespace (str):
+    """
+
+    def __init__(self, name, namespace):
+        self.name = name
+        self.namespace = namespace
+
+
+@pytest.fixture(scope="class")
+def transport_server_setup(
+        request, kube_apis, test_namespace
+) -> TransportServerSetup:
+    """
+    Prepare Transport Server Example.
+
+    :param request: internal pytest fixture to parametrize this method:
+        {example: virtual-server|virtual-server-tls|..., app_type: simple|split|...}
+        'example' is a directory name in TEST_DATA,
+        'app_type' is a directory name in TEST_DATA/common/app
+    :param kube_apis: client apis
+    :param test_namespace:
+    :return: TransportServerSetup
+    """
+    print(
+        "------------------------- Deploy Transport Server Example -----------------------------------"
+    )
+
+    # deploy global config
+    global_config_file = f"{TEST_DATA}/{request.param['example']}/standard/global-configuration.yaml"
+    gc_resource = create_gc(kube_apis.custom_objects, global_config_file, "nginx-ingress")
+
+    # deploy dns
+    dns_file = f"{TEST_DATA}/{request.param['example']}/standard/dns.yaml"
```

Comment:
do we always need to deploy a DNS app? 
for status check, we don't need really to check the traffic flow `client->NGINX->backend`. so I wonder whether we should deploy the example app conditionally

Reply:
I think we do. I tested it locally, removing the dns server but the tests don't pass as the pods never become ready. I think this is because nginx requires a valid listener ? So without the dns service, it doesn't start.

Reply:
we need the app related listeners in GlobalConfiguration, but not the app itslef (its deployment and service)
In that case, the IC will generate the NGINX config with zero upstream servers, which is legit. For example, for TCP TS:
```nginx
upstream ts_default_dns-tcp_dns-app {
    zone ts_default_dns-tcp_dns-app 256k;
    random two least_conn;
}
server {
    listen 5353;

    status_zone dns-tcp;
    proxy_pass ts_default_dns-tcp_dns-app;
    proxy_timeout 10m;
    proxy_connect_timeout 60s;
}
```
the status will be:
Status:
  Message:  Configuration for default/dns-tcp was added or updated
  Reason:   AddedOrUpdated
  State:    Valid
```